File: app/src/main/java/com/guideapp/guideapp/ui/views/map/MapContract.java
package com.guideapp.guideapp.ui.views.map;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v4.app.LoaderManager;
import android.widget.ImageView;

import com.guideapp.guideapp.model.Local;

import java.util.List;

interface MapContract {
    interface View {
        void showLocals(List<Local> locals);

        void showLocalSummary(Local local);

        void showLocalDetailUi(Local local, ImageView view);

        Context getContext();
    }

    interface Presenter {
        void loadLocals(LoaderManager loaderManager);

        void openLocalSummary(Local local);

        void openLocalDetails(@NonNull Local local, ImageView view);
    }
}

File: app/src/main/java/com/guideapp/ui/views/map/MapContract.kt
package com.guideapp.ui.views.map

import android.content.Context
import android.support.v4.app.LoaderManager
import android.widget.ImageView

import com.guideapp.model.Local

internal interface MapContract {
    interface View {
        fun showLocals(locals: List<Local>)

        fun showLocalSummary(local: Local)

        fun showLocalDetailUi(local: Local, view: ImageView)

        fun getContext() : Context
    }

    interface Presenter {
        fun loadLocals(loaderManager: LoaderManager)

        fun openLocalSummary(local: Local?)

        fun openLocalDetails(local: Local, view: ImageView)
    }
}
